### ISSUE TITLE
Skid Row Cracktro 12

### ISSUE THREAD

**Issue by chris70c** [dirkwhoffmann/vAmiga#528]
Bitplane garbage all around...

![vamiga](https://user-images.githubusercontent.com/3858768/133026089-5b4b77c5-e9a9-4f09-b2a0-61b485ef11e2.png)
![winuae](https://user-images.githubusercontent.com/3858768/133026090-dc5f49c0-e4cb-4feb-97d5-68d5609a59a5.png)
[Skidrow.crack12.zip](https://github.com/dirkwhoffmann/vAmiga/files/7151443/Skidrow.crack12.zip)

**Comment by dirkwhoffmann** [comment]
Same reason as #523:

    [569] (112, 62) 040338  0 BCB--- 0000 1070 Agnus:649 XFILES: Trashing BPL5PTL

When compiled with debug option `NO_PTR_DROPS`, it works fine:

<img width="853" alt="Bildschirmfoto 2021-09-13 um 07 45 36" src="https://user-images.githubusercontent.com/12561945/133036709-7101277f-15ed-4f86-a536-3455001ba18c.png">

The current logic is as follows: 

```
template <int x> void
Agnus::setBPLxPTL(u16 value)
{
    trace(BPLREG_DEBUG, "setBPLxPTL(%d, %X)\n", x, value);
    
    // Check if the write collides with DMA
    if (!NO_PTR_DROPS && isBplxEvent(bplEvent[pos.h], x)) {
        
        trace(XFILES, "XFILES: Trashing BPL%dPTL\n", x);
        bplpt[x - 1] = REPLACE_LO_WORD(bplpt[x - 1], 0xFFFE);
        return;
    }

    // Do nothing if pointer was involded in DMA one cycle earlier
    if (dropWrite((BusOwner)(BUS_BPL1 + x - 1))) return;
    
    // Perform the write
    bplpt[x - 1] = REPLACE_LO_WORD(bplpt[x - 1], value & 0xFFFE);
}
```

Either this logic is wrong, or the pointer write is carried out in a wrong DMA cycle, or there should be no bitplane DMA for plane 5 in cycle 62. Needs investigation...

**Comment by dirkwhoffmann** [comment]
All I can say at the moment is that the following fragment from the code shown above is bogus:

    // Check if the write collides with DMA
    if (!NO_PTR_DROPS && isBplxEvent(bplEvent[pos.h], x)) {
        
        trace(XFILES, "XFILES: Trashing BPL%dPTL\n", x);
        bplpt[x - 1] = REPLACE_LO_WORD(bplpt[x - 1], 0xFFFE);
        return;
    }

It was added in August 2020:

<img width="1155" alt="Bildschirmfoto 2021-09-13 um 11 45 04" src="https://user-images.githubusercontent.com/12561945/133070815-ecf788b6-aca4-48c2-ad25-ea92c32c4f57.png">

There must have been some reason why I've added the code, but I don't remember anything specific. If somebody knows "what I did last summer", please give me a hint.

**Comment by dirkwhoffmann** [comment]
Fixed by commit f5474b7

**Comment by dirkwhoffmann** [comment]
Fixed in vAmiga 1.1b2
